ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjector.java

```diff
@@ -266,7 +267,24 @@ private static CreateSource addSchemaFields(
     final TableElements elements = buildElements(preparedStatement, keySchema, valueSchema);
 
     final CreateSource statement = preparedStatement.getStatement();
-    return statement.copyWith(elements, statement.getProperties());
+    final CreateSourceProperties properties = statement.getProperties();
+
+    Optional<String> avroKeySchemaName = Optional.empty();
+    Optional<String> avroValueSchemaName = Optional.empty();
+
+    // Only populate avro key and value schema names when schema ids are explicitly provided
+    // and fetched schema is avro format
```

Comment:
Why only AVRO but not Protobuf/JSON_SR? (Just for my own education.)

Reply:
Good question. let me check whether protobuf and json_sr also require this to be backward compatible. ksqlDB doesn't populate name for protobuf and json_sr currently not sure the reason...

Reply:
protobuf and json_sr also needs name to be same to be backward compatible. See https://github.com/confluentinc/ksql/issues/7925. Currently, ksqlDb doesn't provide a way to set it and default is `ConnectDefault1`...